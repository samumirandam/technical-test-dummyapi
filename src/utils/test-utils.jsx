import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import reduxThunk from "redux-thunk";
import { configureStore as configureReduxStore } from "@reduxjs/toolkit";
import { BrowserRouter } from "react-router-dom";

import reducer from "@reducers";

const middlewares = [reduxThunk];
const mockStore = configureStore(middlewares);

const AllTheProviders = ({ children }) => {
  const store = configureReduxStore({
    reducer: reducer,
  });
  const setupStore = { ...store.getState(), ...children.props };
  const compomentStore = mockStore(setupStore);
  return (
    <Provider store={compomentStore}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
