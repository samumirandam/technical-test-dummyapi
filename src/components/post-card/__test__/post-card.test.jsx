import React from "react";
import { render, screen, fireEvent } from "@utils/test-utils";

import PostCard from "../index";

const defaultProps = {
  owner: {
    id: "60d0fe4f5311236168a109ca",
    title: "ms",
    firstName: "Sara",
    lastName: "Andersen",
    picture: "https://randomuser.me/api/portraits/women/58.jpg",
  },
  text: "test text",
  openUserProfile: jest.fn(),
};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<PostCard {...setupStore} />, {});
};

describe("Test for PostCard component", () => {
  test("Should render without errors", () => {
    setup();
    expect(screen.getByText("test text")).toBeTruthy();
  });

  test("Should renders the same component", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test("Should render click in open user profile", () => {
    setup();
    fireEvent.click(screen.getByText("Sara Andersen"));
    expect(defaultProps.openUserProfile).toHaveBeenCalled();
  });
});
