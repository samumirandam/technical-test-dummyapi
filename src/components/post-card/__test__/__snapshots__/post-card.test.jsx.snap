// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for PostCard component Should renders the same component 1`] = `
<div>
  <div
    class="PostCard"
  >
    <div
      class="Button flat"
    >
      <div
        class="UserImage"
      >
        <figure>
          <img
            alt="Image Sara Andersen"
            src="https://randomuser.me/api/portraits/women/58.jpg"
          />
        </figure>
        <p>
          Sara Andersen
        </p>
      </div>
    </div>
    <figure
      class="PostCard__image"
    >
      <img
        alt="test text"
      />
    </figure>
    <div
      class="PostCard__icons"
    >
      <div
        class="Button PostCard__tag flat"
      >
        <div
          class="Icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.88 4.78a3.489 3.489 0 0 0-.37-.9 3.24 3.24 0 0 0-.6-.79 3.78 3.78 0 0 0-1.21-.81 3.74 3.74 0 0 0-2.84 0 4 4 0 0 0-1.16.75l-.05.06-.65.65-.65-.65-.05-.06a4 4 0 0 0-1.16-.75 3.74 3.74 0 0 0-2.84 0 3.78 3.78 0 0 0-1.21.81 3.55 3.55 0 0 0-.97 1.69 3.75 3.75 0 0 0-.12 1c0 .317.04.633.12.94a4 4 0 0 0 .36.89 3.8 3.8 0 0 0 .61.79L8 14.31l5.91-5.91c.237-.233.44-.5.6-.79A3.578 3.578 0 0 0 15 5.78a3.747 3.747 0 0 0-.12-1zm-1 1.63a2.69 2.69 0 0 1-.69 1.21l-5.21 5.2-5.21-5.2a2.9 2.9 0 0 1-.44-.57 3 3 0 0 1-.27-.65 3.25 3.25 0 0 1-.08-.69A3.36 3.36 0 0 1 2.06 5a2.8 2.8 0 0 1 .27-.65c.12-.21.268-.4.44-.57a2.91 2.91 0 0 1 .89-.6 2.8 2.8 0 0 1 2.08 0c.33.137.628.338.88.59l1.36 1.37 1.36-1.37a2.72 2.72 0 0 1 .88-.59 2.8 2.8 0 0 1 2.08 0c.331.143.633.347.89.6.174.165.32.357.43.57a2.69 2.69 0 0 1 .35 1.34 2.6 2.6 0 0 1-.06.72h-.03z"
            />
          </svg>
        </div>
      </div>
      <div
        class="Button flat"
      >
        <div
          class="Icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14.5 2h-13l-.5.5v9l.5.5H4v2.5l.854.354L7.707 12H14.5l.5-.5v-9l-.5-.5zm-.5 9H7.5l-.354.146L5 13.293V11.5l-.5-.5H2V3h12v8z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="PostCard__tags"
    />
    <p
      class="PostCard__text"
    >
      test text
    </p>
  </div>
</div>
`;
